pkg/helm/client/client.go

```diff
@@ -98,16 +100,19 @@ func NewRESTClientGetter(mgr manager.Manager, ns string) (genericclioptions.REST
 
 var _ kube.Interface = &ownerRefInjectingClient{}
 
-func NewOwnerRefInjectingClient(base kube.Client, ownerRef metav1.OwnerReference) kube.Interface {
+func NewOwnerRefInjectingClient(base kube.Client, restMapper meta.RESTMapper,
+	cr *unstructured.Unstructured) kube.Interface {
 	return &ownerRefInjectingClient{
-		refs:   []metav1.OwnerReference{ownerRef},
-		Client: base,
+		Client:     base,
+		restMapper: restMapper,
+		owner:      cr,
```

Comment:
Should we validate that the owner has GVK + name set here?

Reply:
@joelanford Let me know what you think as well, on whether we should validate.

Reply:
For all practical purposes, I don't think we _need_ to check because:
- `SupportsOwnerReference()` will return an error if the GVK doesn't have a valid rest mapping.
- In the context of the helm operator, the owner's `name` and `uid` will have been populated by `client.Get`.

BUT from the perspective of using this `ownerRefInjectingClient` independently, it would probably make sense to verify those fields. I'm good with verifying the owner's GVK, name, and uid as part of the constructor (and updating the signature to also include returning an error).